src/pages/learn/validation.mdx

```diff
@@ -155,10 +95,65 @@ a separate fragment:
 }
 ```
 
-This has just scratched the surface of the validation system; there
-are a number of validation rules in place to ensure that a GraphQL query
-is semantically meaningful. The specification goes into more detail about this
-topic in the "Validation" section, and the
-[validation](https://github.com/graphql/graphql-js/blob/main/src/validation)
-directory in GraphQL.js contains code implementing a
-specification-compliant GraphQL validator.
+### Cyclic fragment spreads
+
+To start, let's take a complex valid query. This is a nested query, but with the duplicated fields factored out into a fragment:
+
+```graphql
+# { "graphiql": true }
+query {
+  hero {
+    ...NameAndAppearances
+    friends {
+      ...NameAndAppearances
+      friends {
+        ...NameAndAppearances
+      }
+    }
+  }
+}
+
+fragment NameAndAppearances on Character {
+  name
+  appearsIn
+}
+```
+
+And this query is valid. Let's take a look at some invalid queries...
+
+A fragment cannot refer to itself or create a cycle, as this could result in an unbounded result! Here's the same query above but without the explicit three levels of nesting:
```

Comment:
We mentioned the query was valid already and we only have one query below here; here's an attempt to merge these paragraphs:

```suggestion
The following is an alternative to the above query, attempting to use recursion instead of the explicit three levels of nesting. This new query is invalid because a fragment cannot refer to itself (directly or indirectly) since the resulting cycle could create an unbounded result!
```